hazelcast/src/main/java/com/hazelcast/partition/impl/ForceUpdateSyncWaitingReplicaVersionsOperation.java

```diff
@@ -0,0 +1,128 @@
+package com.hazelcast.partition.impl;
+
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.partition.InternalPartition;
+import com.hazelcast.partition.InternalPartitionLostEvent;
+import com.hazelcast.partition.InternalPartitionService;
+import com.hazelcast.partition.MigrationCycleOperation;
+import com.hazelcast.spi.AbstractOperation;
+import com.hazelcast.spi.PartitionAwareOperation;
+import com.hazelcast.spi.PartitionAwareService;
+import com.hazelcast.spi.impl.NodeEngineImpl;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+// Runs locally when the node becomes owner of a partition upon a node failure
+// Finds the replica indices that are on the sync-waiting state. Those indices represents the lost backups of the partition.
+// Therefore, it publishes InternalPartitionLostEvent objects to notify related services
+// It also updates the version for lost replicas to the first available version value after the lost backups, or 0 if N/A
+final class ForceUpdateSyncWaitingReplicaVersionsOperation
```

Comment:
This operation and `PromoteFromBackupOperation` are both executed when a node becomes owner of a partition. What about merging both to a single operation? Also  `PromoteFromBackupOperation` contains some logic from this operation, see `setMissingReplicaVersions()`, which I borrowed from your branch before...